pipeline/recorder/src/test/scala/uk/ac/wellcome/platform/recorder/fixtures/WorkerServiceFixture.scala

```diff
@@ -1,46 +1,111 @@
+  class BrokenMemoryRecorderVhs extends MemoryRecorderVhs() {
+    override def put(id: Version[String, Int])(entry: Entry): WriteEither =
+      Left(StoreWriteError(new Error("BOOM!")))
+  }
+
+  // Cache these here to improve compilation times (otherwise they are
+  // re-derived every time they are required)
+  implicit val workEncoder: Encoder[TransformedBaseWork] = deriveEncoder
+  implicit val workDecoder: Decoder[TransformedBaseWork] = deriveDecoder
+  implicit val workCodec: Codec[TransformedBaseWork] = Codec.typeCodec
+
+  def withRecorderVhs[R](testWith: TestWith[RecorderVhs, R]): R = {
+    testWith(new MemoryRecorderVhs())
+  }
+
+  def withBrokenRecorderVhs[R](testWith: TestWith[RecorderVhs, R]): R = {
+    testWith(new BrokenMemoryRecorderVhs())
+  }
+
+  def withMemoryMessageSender[R](
+    testWith: TestWith[MemoryMessageSender, R]): R = {
+    testWith(new MemoryMessageSender())
+  }
+
+  def withWorkerService[R](queue: Queue,
+                           vhs: RecorderVhs,
+                           msgSender: MemoryMessageSender)(
+    testWith: TestWith[RecorderWorkerService[String], R]): R =
     withActorSystem { implicit actorSystem =>
-      withMetricsSender() { metricsSender =>
-        withSNSWriter(topic) { snsWriter =>
-          withTypeVHS[TransformedBaseWork, EmptyMetadata, R](
-            bucket = storageBucket,
-            table = table) { versionedHybridStore =>
-            withMessageStream[TransformedBaseWork, R](
-              queue = queue,
-              metricsSender = metricsSender) { messageStream =>
-              val workerService = new RecorderWorkerService(
-                versionedHybridStore = versionedHybridStore,
-                messageStream = messageStream,
-                snsWriter = snsWriter
-              )
-
-              workerService.run()
-
-              testWith(workerService)
-            }
-          }
-        }
+      implicit val streamStore =
+        MemoryTypedStoreCompanion[ObjectLocation, TransformedBaseWork]()
+      withMessageStream[TransformedBaseWork, R](queue = queue) { msgStream =>
+        val workerService = new RecorderWorkerService(vhs, msgStream, msgSender)
+        workerService.run()
+        testWith(workerService)
       }
     }
+
+  def assertWorkStored[T <: TransformedBaseWork](
+    vhs: RecorderVhs,
+    work: T,
+    expectedVhsVersion: Int = 0): Version[String, Int] = {
+
+    val id = work.sourceIdentifier.toString
+    vhs.getLatest(id) shouldBe
+      Right(
+        Identified(
+          Version(id, expectedVhsVersion),
+          HybridStoreEntry(work, EmptyMetadata())))
+    Version(id, expectedVhsVersion)
+  }
+
+  def assertWorkNotStored[T <: TransformedBaseWork](vhs: RecorderVhs,
+                                                    work: T) = {
+
+    val id = work.sourceIdentifier.toString
+    val workExists = vhs.getLatest(id) match {
+      case Left(_)  => false
+      case Right(_) => true
+    }
+    workExists shouldBe false
```

Comment:
Possibly simpler:

```scala
vhs.getLatest(id) shouldBe a[Left[_]]
```

And I think that will also print out the `Right` (and thus the location/work) if it fails?

Reply:
Tried something like this originally but couldn't get it working for some reason, managed to do it now, cheers!